utils/common/src/main/java/org/apache/brooklyn/util/collections/CollectionHelpers.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.util.collections;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class CollectionHelpers {
+    private static final Logger log = LoggerFactory.getLogger(CollectionHelpers.class);
+
+    /**
+     * Does a shallow merge of two maps.
+     */
+    public static Map<String, Object> mergeMapsAndTheirInnerMapValues(Map<String, Object> defaultEntries, Map<String, Object> overrideWith) {
```

Comment:
This is the sort of method that it's really easy to write great test coverage for. We should do that, including where default entry is of different type than overriding entry etc.

It's unclear what a user would expect, when merging maps. Why stop at one level down? What about merging or overriding lists?